File: backend/app/schemas/attachment.py
"""
Attachment schemas.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel


class AttachmentResponse(BaseModel):
    id: uuid.UUID
    message_id: uuid.UUID | None
    original_filename: str
    stored_filename: str
    file_path: str
    mime_type: str
    file_size: int
    attachment_type: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AttachmentUploadResponse(BaseModel):
    attachments: list[AttachmentResponse]

File: frontend/src/types/attachment.ts
export type AttachmentType =
  | "image"
  | "generated_image"
  | "video"
  | "pdf"
  | "code"
  | "text"
  | "csv"
  | "formula"
  | "other";

export interface Attachment {
  id: string;
  message_id: string | null;
  original_filename: string;
  stored_filename: string;
  file_path: string;
  mime_type: string;
  file_size: number;
  attachment_type: AttachmentType;
  created_at: string;
}

export interface AttachmentUploadResponse {
  attachments: Attachment[];
}
